File: VB/Models/CustomAppointmentTemplateContainer.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.Web.Mvc
Imports DevExpress.Web.ASPxScheduler

Namespace SchedulerSimpleCustomFormMvc.Models
	Public Class CustomAppointmentFormTemplateContainer
		Inherits AppointmentFormTemplateContainer
		Public Sub New(ByVal scheduler As MVCxScheduler)
			MyBase.New(scheduler)
		End Sub
		Public ReadOnly Property Price() As Decimal?
			Get
'INSTANT VB NOTE: The local variable price was renamed since Visual Basic will not allow local variables with the same name as their enclosing function or property:
				Dim price_Renamed As Object = Appointment.CustomFields("Price")
				Return If(price_Renamed Is DBNull.Value, 0, CType(price_Renamed, Decimal?))
			End Get
		End Property
	End Class
End Namespace

File: CS/Models/CustomAppointmentTemplateContainer.cs
using System;
using DevExpress.Web.Mvc;
using DevExpress.Web.ASPxScheduler;

namespace SchedulerSimpleCustomFormMvc.Models {
    public class CustomAppointmentFormTemplateContainer : AppointmentFormTemplateContainer {
        public CustomAppointmentFormTemplateContainer(MVCxScheduler scheduler)
            : base(scheduler) {
        }
        public decimal? Price {
            get {
                object price = Appointment.CustomFields["Price"];
                return price == DBNull.Value ? 0 : (decimal?)price;
            }
        }
    }
}